### ISSUE TITLE
BMG option for Finnish T-26C(r)

### ISSUE THREAD

**Issue by BigAl737** [vasl-developers/vasl#518]
http://www.gamesquad.com/forums/index.php?threads/vasl-map-online-problem-suggestion-form.88223/post-1949829

I should have this done soon. Doug, is it too late to include in the VASL 6.4.3 final release?

Side note: Only 1 in the Hakkaa Paalle counter mix. Need two for scenario 173. Not a problem for vasl of course.

**Comment by BigAl737** [comment]
![fit-26c r bmg](https://user-images.githubusercontent.com/10384413/47683201-c1989080-db83-11e8-9d27-16294d1a4029.png)
fiT-26c(r)bmg.png (addition)

![fit-26c r](https://user-images.githubusercontent.com/10384413/47683204-c52c1780-db83-11e8-9129-3905502e8bb6.png)
fiT-26c(r).png (replacement)

![fit-26c r b](https://user-images.githubusercontent.com/10384413/47683211-c9f0cb80-db83-11e8-9fb3-fd6b1ead5ef6.png)
fiT-26c(r)B.png (replacement)

I'll see if I can get these loaded where they need to go for the next vasl release. Over a foot of snow today that's dying to be skied on :-) or shoveled :-(

**Comment by BigAl737** [comment]
I "_think_" I created a new branch and uploaded the counters to the appropriate spot. Doug, please check me when you have time. If all is okay then let me know what the next step is.

I'm not sure how to load the new counter (fiT-26c(r)bmg.png) to VASLs counter window. It should go in the same window as the non-bmg version. Doug, please do this for me or better yet, if you have time, teach me. Thanks!

**Comment by BigAl737** [comment]
Or maybe these are better?

![fit-26c r bmg-altversion](https://user-images.githubusercontent.com/10384413/47765345-67c8c100-dc7e-11e8-9c20-9703b71c8c36.png)

![fit-26c r -altversion](https://user-images.githubusercontent.com/10384413/47765411-aa8a9900-dc7e-11e8-96a6-b0424b631ec6.png)

![fit-26c r b-altversion](https://user-images.githubusercontent.com/10384413/47765346-6a2b1b00-dc7e-11e8-9a49-19a3ee761594.png)

**Comment by derimmer** [comment]
Al, I will check this out as you request.

**Comment by BigAl737** [comment]
I never thanked you Doug...Thanks!

Looks like Sam got it done for me. Thanks Sam.

I'd like to learn how to update the build file. Brian or Zoltan, care to pass on any instructions?

**Comment by BigAl737** [comment]
I think we can close this issue.

**Comment by BigAl737** [comment]
Doug, looks like these are already in place for you to roll into your upcoming release.

**Comment by derimmer** [comment]
Super. Thanks Al.

**Comment by BigAl737** [comment]
Pushed to the fiT-26C(r)Counter branch. How'd I do Doug? Hope I didn't do anything you can't undo.

**Comment by derimmer** [comment]
@BigAl737 I see that you have changed the buildfile. Which is ok. Did you use the v6.4.4 buildfile or an earlier version? When changing the buildfile, I think you always need to start with the latest version as GitHub replaces the entire file. Code files are different. 

With GitHub damn near everything can be undone. For next steps, see my comments on the Flak30counter issue

**Comment by derimmer** [comment]
Looked at your commit in more detail. You may have a problem because the branch you used was not up to date with develop. There are several ways to correct this. We will do it the easy way. I would suggest you do the Flak30Counter pull request but on the final screen, do NOT delete your Flak30counter branch.

**Comment by BigAl737** [comment]
Thanks Doug. I did modify the build file from VASL 6.4.4. I kept the modified buildfile on my local system so I can push it again. Thanks for your help! I really appreciate it. I'm outta time for today so will get back to this in a few days. Just glad I didn't harm anything. I'm working hard to understand how to push these fixes so it eases your workload. Again...much appreciated!

**Comment by BigAl737** [comment]
Added both T-26 versions to the build file. Corrections confirmed by Doug. Closing.

**Comment by derimmer** [comment]
Hi

 

I recently received this message. The key part is at the end. . . he wants to repurpose the vasl counter images to set up VASL scenarios. 

 

I know there is some sensitivity with respect to counter and map images. 

 

Any thoughts about this request?

 

Doug 

 

From: GameSquad Forums <support@gamesquad.com> 
Sent: Sunday, April 14, 2019 2:52 AM
To: DougRim <dougrimmer@rogers.com>
Subject: Pacman Ghost started a conversation with you: "Re-using VASL counter images"

 



 <http://www.gamesquad.com/forums/index.php> GameSquad Forums 


Pacman Ghost started a new conversation with you at  <http://www.gamesquad.com/forums/index.php> GameSquad Forums.


 <http://www.gamesquad.com/forums/index.php?conversations/re-using-vasl-counter-images.59050/unread> Re-using VASL counter images


Hi

You may be aware that I've been working on a program called vasl-templates, that lets people set up VASL scenarios. If not, you can find out about it here:

*	 <http://www.gamesquad.com/forums/index.php?threads/new-program-to-help-set-up-vasl-scenarios.148281/> http://www.gamesquad.com/forums/index.php?threads/new-program-to-help-set-up-vasl-scenarios.148281/
*	 <https://rawgit.com/pacman-ghost/vasl-templates/master/vasl_templates/webapp/static/help/index.html?tab=userguide> https://rawgit.com/pacman-ghost/vasl-templates/master/vasl_templates/webapp/static/help/index.html?tab=userguide

One of the things it does is extract (at runtime) the images for vehicles/ordnance from VASL's .vmod file, and serves them up via a localhost web server i.e. in a label in VASSAL, there is HTML of the form:



<img src="http://localhost:5010/counter/1234/front">

and vasl-templates is listening on port 5010 to serve up the image for GPID 1234.

This works well e.g. the thing on the right in the screenshot below is a label in VASSAL, with the counter images dynamically inserted by vasl-templates:
  <https://raw.githubusercontent.com/pacman-ghost/vasl-templates/master/vasl_templates/webapp/static/help/images/ob_vehicles.png> 
but it makes sharing the scenario difficult, since both players need to be running the vasl-templates program (in order to get the localhost web server that serves the counter images).

I'd like to remedy this by putting the counter images online i.e. the VASSAL labels will now look something like this:



<img src="http://mydomain.com/counter/1234/front">

and the images will be available, even if neither player is running my program. Note that the images will be pre-extracted from the .vmod file, and served as static files i.e. there won't be any code running on the online web server.

I need to look at the licensing more closely, but it would seem to be OK - VASL is LGPL, vasl-templates is AGPL - but I just wanted to run it by you guys first, and make sure you were OK with me doing this.

Thanks


 <http://www.gamesquad.com/forums/index.php?conversations/re-using-vasl-counter-images.59050/unread> View this conversation 

 <http://www.gamesquad.com/forums/index.php?conversations/> View all your conversations 

Please do not reply to this message. You must visit the forum to reply.

This message was sent to you because your preferences are set to receive email when a new conversation message is received.  <http://www.gamesquad.com/forums/index.php?email-stop/25637/conversation&c=3e1816ced64d37466e54e2cca1bd4a33> Stop receiving these emails.


 <http://www.gamesquad.com/forums/index.php> Visit GameSquad Forums

**Comment by BigAl737** [comment]
I don't have issue with the use of images. I wonder about quality control. Is he referencing the wreck counter "has info" trait to populate the back of the counter notes? If not, there's a good chance for errors. If so, sometimes wreck counters are within the hidden ? counter menu and sometimes they're individually defined within the "replace with other trait". I would like to keep the option for both.

**Comment by davidsullivan317** [comment]
The images are already online: https://github.com/vasl-developers/vasl/tree/develop/dist/images 

He should just point his software there and not copy them somewhere else. 

We've had the discussion re use of the images a number of times and it's okay to use them as long as you comply with the LGPL.

**Comment by hongkongwargamer** [comment]
I was told explicitly when I started my blog (which uses all manner of VASL images) that I should ask for permission from Rodney Kinney, and I did.  

This is the notice on VASL HELP file .. "The VASSAL Engine is supported by an open-source project founded and administered by Rodney Kinney and distributed under the GNU Library General Public License (LGPL). This distribution includes a number of third-party binary libraries, distributed under terms of their respective licenses.  Artwork and other media contained in module files are the property of their respective owners.  The VASSAL web site administrators will comply with any request to remove copyrighted material from the web site."

I wonder if material shared on the GitHub site constitutes something akin to a Creative Commons licence where usage for a COMMERCIAL purpose requires a more explicit permission?

Plus - remember some of the images that we use for counters are **not** created from scratch.  Allowing them to be used further in other products might create an issue that will turn into a backlash for VASL?
